src/iuse.cpp

```diff
@@ -5157,17 +5156,16 @@ static bool heat_item( Character &p )
         add_msg( m_info, _( "Never mind." ) );
         return false;
     }
-    item *target = heat->get_food();
     // simulates heat capacity of food, more weight = longer heating time
     // this is x2 to simulate larger delta temperature of frozen food in relation to
     // heating non-frozen food (x1); no real life physics here, only approximations
-    int duration = to_turns<int>( time_duration::from_seconds( to_gram( target->weight() ) ) ) * 10;
-    if( target->has_own_flag( flag_FROZEN ) && !target->has_flag( flag_EATEN_COLD ) ) {
+    int duration = to_turns<int>( time_duration::from_seconds( to_gram( heat->weight() ) ) ) * 10;
```

Comment:
Sort of off-topic:

Food is very water-based, but if we heat up things like aluminium, it should be like 5 times less energy-intensive relative to weight.
(4,7 times according to [this src](https://gchem.cm.utexas.edu/data/section2.php?target=heat-capacities.php))

Would it be worth it to make PR about heat capacity?

Reply:
That calculation is nonsense anyways. It does not take either temperature, heat capacity or latent heat of fusion incto account at all.

It should be somthing like

`duration = ( target_temperature - item_temperature ) * heat_capacity * A * mass  `
And if it is frozen->liquid transition then add `mass*latent_heat_fusion` to above.

`A` is some constant to present how fast heat flows into the item.